recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -0,0 +1,235 @@
+        self._cmake.definitions["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = "TRUE"
+        self._cmake.definitions["ENABLE_SSL"] = self._ssl_cmake_value
+        self._cmake.definitions["ENABLE_SASL"] = self._sasl_cmake_value
+        self._cmake.definitions["ENABLE_STATIC"] = "OFF" if self.options.shared else "ON"
+        self._cmake.definitions["ENABLE_TESTS"] = "OFF"
+        self._cmake.definitions["ENABLE_EXAMPLES"] = "OFF"
+        self._cmake.definitions["ENABLE_SRV"] = "ON" if self.options.srv else "OFF"
+        self._cmake.definitions["ENABLE_MAINTAINER_FLAGS"] = "OFF"
+        self._cmake.definitions["ENABLE_AUTOMATIC_INIT_AND_CLEANUP"] = "ON"
+        self._cmake.definitions["ENABLE_CRYPTO_SYSTEM_PROFILE"] = "OFF"
+        self._cmake.definitions["ENABLE_TRACING"] = "OFF"
+        self._cmake.definitions["ENABLE_COVERAGE"] = "OFF"
+        self._cmake.definitions["ENABLE_SHM_COUNTERS"] = "OFF"
+        self._cmake.definitions["ENABLE_MONGOC"] = "ON"
+        self._cmake.definitions["ENABLE_BSON"] = "ON"
+        self._cmake.definitions["ENABLE_SNAPPY"] = "ON" if self.options.with_snappy else "OFF"
+        self._cmake.definitions["ENABLE_ZLIB"] = "SYSTEM" if self.options.with_zlib else "OFF"
+        self._cmake.definitions["ENABLE_ZSTD"] = "ON" if self.options.with_zstd else "OFF"
+        self._cmake.definitions["ENABLE_MAN_PAGES"] = False
+        self._cmake.definitions["ENABLE_HTML_DOCS"] = False
+        self._cmake.definitions["ENABLE_EXTRA_ALIGNMENT"] = True
+        self._cmake.definitions["ENABLE_RDTSCP"] = False
+        self._cmake.definitions["ENABLE_APPLE_FRAMEWORK"] = False
+        self._cmake.definitions["ENABLE_ICU"] = "ON" if self.options.with_icu else "OFF"
+        self._cmake.definitions["ENABLE_UNINSTALL"] = False
+        self._cmake.definitions["ENABLE_CLIENT_SIDE_ENCRYPTION"] = "OFF" # libmongocrypt recipe not yet in CCI
+        self._cmake.definitions["ENABLE_MONGODB_AWS_AUTH"] = "AUTO"
+        self._cmake.definitions["ENABLE_PIC"] = self.options.get_safe("fPIC", True)
+        if self.options.with_ssl == "openssl":
+            self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info["openssl"].rootpath
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="THIRD_PARTY_NOTICES", dst="licenses", src=self._source_subfolder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "mongoc-1.0"
```

Comment:
```suggestion
        # FIXME: two config files should be generated (mongoc-1.0-config.cmake and bson-1.0-config.cmake),
        # but it can't be modeled right now.
        self.cpp_info.filenames["cmake_find_package"] = "mongoc-1.0"
```

To not forget to fix that if conan client improves `cmake_find_package` and `cmake_find_package_multi` generators in the future.